app/src/main/java/com/arctouch/codechallenge/home/HomeViewModel.java

```diff
@@ -0,0 +1,224 @@
+package com.arctouch.codechallenge.home;
+
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+import android.text.TextUtils;
+
+import com.arctouch.codechallenge.api.TmdbApi;
+import com.arctouch.codechallenge.api.TmdbApiSettings;
+import com.arctouch.codechallenge.data.Cache;
+import com.arctouch.codechallenge.model.Genre;
+import com.arctouch.codechallenge.model.GenreResponse;
+import com.arctouch.codechallenge.model.Movie;
+import com.arctouch.codechallenge.model.UpcomingMoviesResponse;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+import retrofit2.Response;
+
+/**
+ * Created by eduardocucharro on 25/03/18.
+ */
+
+public class HomeViewModel extends ViewModel {
+
+    private final Executor executor;
+    private final TmdbApi api;
+    private final Cache cache;
+    private long page = 0;
+    private boolean noMoreMovies = false;
+    private MutableLiveData<List<Movie>> movieList = new MutableLiveData<>();
+    private MutableLiveData<Boolean> loading = new MutableLiveData<>();
+    private MutableLiveData<Throwable> errorState = new MutableLiveData<>();
+    private List<Movie> movies = new ArrayList<>();
+    private List<Movie> searchMovieList = new ArrayList<>();
+    private long searchPage = 0;
+    private boolean noMoreSearch = false;
+    private String query;
+
+    public MutableLiveData<List<Movie>> getMovieList() {
+        return movieList;
+    }
+
+    public MutableLiveData<Boolean> getLoading() {
+        return loading;
+    }
+
+    public MutableLiveData<Throwable> getErrorState() {
+        return errorState;
+    }
+
+    public HomeViewModel() {
```

Comment:
Its better remove this right?

Reply:
Right, I did it as an temporary alternative until apply dependency injection!